src/encode.jl

```diff
@@ -0,0 +1,47 @@
+"""
+    encode(img; kwargs...)
+
+Encode image `img` into JPEG byte sequences.
+"""
+function encode(img::AbstractMatrix; transpose=false, kwargs...)
+    # quantilized into 8bit sequences first
+    AT = Array{n0f8(eltype(img)), ndims(img)}
+    # jpegturbo is a C library and assumes row-major memory order, thus `collect` the data into
+    # contiguous memeory layout already makes a transpose.
+    img = transpose ? convert(AT, img) : convert(AT, PermutedDimsArray(img, (2, 1)))
+
+    return _encode(img; kwargs...)
+end
+
+function _encode(img::Matrix{Gray{N0f8}}; quality=100)
+    cinfo = Ref{LibJpeg.jpeg_compress_struct}()
+    cinfo_ptr = Base.unsafe_convert(Ptr{LibJpeg.jpeg_compress_struct}, cinfo)
```

Comment:
When you use the result of `unsafe_xxx` without `GC.@preserve`-ing the corresponding  Julia object, it's basically GC unsafe.